File: football_field_manager_demo/lib/data/models/request/booking_request.dart
class BookingRequest {
  String? userPhoneNumber;
  int? footballFieldId;
  String? bookingDate;
  int? timeSlotId;
  int? pitchNumber;
  List<int>? serviceIds;
  int? price;

  BookingRequest(
      {this.userPhoneNumber,
        this.footballFieldId,
        this.bookingDate,
        this.timeSlotId,
        this.pitchNumber,
        this.serviceIds,
        this.price});

  BookingRequest.fromJson(Map<String, dynamic> json) {
    userPhoneNumber = json['userPhoneNumber'];
    bookingDate = json['bookingDate'];
    footballFieldId = json['footballFieldId'];
    timeSlotId = json['timeSlotId'];
    pitchNumber = json['pitchNumber'];
    serviceIds = json['serviceIds'].cast<int>();
    price = json['price'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['userPhoneNumber'] = this.userPhoneNumber;
    data['bookingDate'] = this.bookingDate;
    data['footballFieldId'] = this.footballFieldId;
    data['timeSlotId'] = this.timeSlotId;
    data['pitchNumber'] = this.pitchNumber;
    data['serviceIds'] = this.serviceIds;
    data['price'] = this.price;
    return data;
  }
}

File: football_field_manager/src/main/java/com/example/football_field_manager/dto/request/BookingRequest.java
package com.example.football_field_manager.dto.request;

import com.example.football_field_manager.constant.BookingStatus;
import com.example.football_field_manager.constant.PaymentStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class BookingRequest {
    String orderCode = "ORD" + System.currentTimeMillis();

    LocalDate bookingDate;

    String userPhoneNumber;

    Integer pitchNumber;

    Long footballFieldId;

    Long timeSlotId;

    Set<Long> serviceIds = new HashSet<>();

    BookingStatus bookingStatus = BookingStatus.pending;

    PaymentStatus paymentStatus = PaymentStatus.pending;

    BigDecimal price;
}
